docs/tech_specs/dup_v2_alert_widget_backend.md

```diff
@@ -0,0 +1,141 @@
+
+```
+first_section_alerts =
+    opts
+    |> Alert.fetch()
+    |> Enum.filter(&relevant?/1)
+    |> choose_alert()
+
+second_section_alerts = []
+if length(sections) == 2 do
+    second_section_alerts =
+        opts
+        |> Alert.fetch()
+        |> Enum.filter(&relevant?/1)
+        |> choose_alert()
+end
+
+alerts = first_section_alerts ++ second_section_alerts
+```
+
+If screen is configured with only one section, `alerts` will contain up to one item. If there are two sections configured, `alerts` will contain up to two items.
+
+With this list, we create three `WidgetInstance` objects for each rotation:
+
+```
+[
+  %DupAlert{screen: config, rotation_index: 0, alerts: alerts},
+  %DupAlert{screen: config, rotation_index: 1, alerts: alerts},
+  %DupAlert{screen: config, rotation_index: 2, alerts: alerts}
+]
+```
+
+### WidgetInstance
+
+In `WidgetInstance.DupAlert`, the following type will be used for the struct:
+
+```
+@type t :: %__MODULE__{
+          screen: Screen.t(),
+          alerts: [Alert.t()] | [Alert.t(), Alert.t()],
+          rotation_index: 0 | 1 | 2
+        }
+```
+
+\*Although the struct can take a list of two alerts, only the first alert in the list will ever be displayed on the screen.
+
+Because irrelevant alerts are filtered out in the `CandidateGenerator`, we are able to always return `true` from `valid_candidate?/1`.
+
+A function `get_region_and_headsign(alert, parent_stop_id)` will be created based on the logic from [this function](/lib/screens/dup_screen_data/data.ex#L11). This function will be used by `serialize/1` to determine an alert's region and headsign. If `region == :inside`, `headsign` is `nil`. If `region == :boundary`, `headsign` is populated using [this map](/config/config.exs#L86).
+
+`slot_names/1` will need to use different logic for each rotation. See spec for details.
+
+Two serialize functions will be created: `serialize_partial_alert/1` and `serialize_takeover_alert/1`.
+
+`serialize_partial_alert/1` returns the following map:
+
+```
+%{
+    alert_text: %FreeTextLine{icon: :warning, text: ...},
+    color: :red | :orange | :green | :blue
+}
+```
+
+`alert_text.text` is either `No ${headsign} trains` or `No ${line} service` depending on the value of `headsign` in the return of `get_region_and_headsign/2`.
+
+`serialize_takeover_alert/1` returns the following map:
+
+```
+%{
+    alert_text: %FreeTextLine{icon: :warning, text: ...},
+    color: :red | :orange | :green | :blue,
+    remedy: %FreeTextLine{icon: :shuttle | nil, text: ...}
+}
+```
```

Comment:
I think this will also need to include values to render the alert header. A nested map might make sense, something like
```ex
%{
  ...
  header: %{
    icon: :logo,
    text: String.t(),
    time: String.t(),
    pattern: :hatched | :x | :chevron,
    color: :red | :orange | :green | :blue | :silver | :purple | :yellow
  }
}
```
(based on what I saw in the existing DUP header component and CSS)

Reply:
Didn't even think about that. Added that and a little blurb about what each value is. Let me know what you think.